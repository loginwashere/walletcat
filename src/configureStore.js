import { createStore, applyMiddleware, compose } from 'redux'
import reducers from './reducers'
import middleware from './middleware'
import { addLocaleData } from 'react-intl'
import { DEFAULT_LOCALE } from './config'
import localeMessages from './translations'

import enLocaleData from 'react-intl/locale-data/en'
import ruLocaleData from 'react-intl/locale-data/ru'
import ukLocaleData from 'react-intl/locale-data/uk'

addLocaleData(enLocaleData)
addLocaleData(ruLocaleData)
addLocaleData(ukLocaleData)

const locale = DEFAULT_LOCALE
const messages = localeMessages[locale]

const initialState = {
  intl: {
    locale,
    messages,
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const configureStore = (preloadedState = {}) => createStore(
  reducers,
  {
    ...preloadedState,
    ...initialState
  },
  composeEnhancers(applyMiddleware(...middleware))
)

export default configureStore
