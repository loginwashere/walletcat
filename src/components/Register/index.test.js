import React from 'react'
import { Provider } from 'react-redux'
import configureStore from '../../configureStore'
import renderer from 'react-test-renderer'
import { Register } from '.'

it('renders without crashing', () => {
  const store = configureStore()
  const tree = renderer.create(
    <Provider store={store}>
      <Register dispatch={store.dispatch} />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
