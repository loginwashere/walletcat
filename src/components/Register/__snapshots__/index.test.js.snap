exports[`test renders without crashing 1`] = `
<form
  className="register__form form-horizontal"
  onSubmit={[Function]}>
  <div
    className="form-group has-feedback">
    <h1>
      Please
    </h1>
    <h2>
      Create an account
    </h2>
    <div />
  </div>
  <div
    className="form-group has-feedback">
    <label
      className="sr-only control-label"
      htmlFor={undefined}>
      Email
    </label>
    <input
      autoFocus={undefined}
      className="form-control"
      id={undefined}
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value="" />
  </div>
  <div
    className="form-group has-feedback">
    <label
      className="sr-only control-label"
      htmlFor={undefined}>
      Username
    </label>
    <input
      autoFocus={undefined}
      className="form-control"
      id={undefined}
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Username"
      required={true}
      type="text"
      value="" />
  </div>
  <div
    className="form-group has-feedback">
    <label
      className="sr-only control-label"
      htmlFor={undefined}>
      Password
    </label>
    <input
      autoFocus={undefined}
      className="form-control"
      id={undefined}
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value="" />
  </div>
  <button
    className="btn btn-lg btn-primary btn-block btn btn-default"
    disabled={true}
    type="submit">
    Create an account
  </button>
  <button
    className="btn btn-lg btn-block btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button">
    Sign Up using Facebook
  </button>
</form>
`;
